src/main/kotlin/com/openlattice/mechanic/upgrades/AddOriginIdToDataPrimaryKey.kt

```diff
@@ -0,0 +1,39 @@
+package com.openlattice.mechanic.upgrades
+
+import com.openlattice.mechanic.Toolbox
+import com.openlattice.postgres.PostgresColumn.ORIGIN_ID
+import com.openlattice.postgres.PostgresDataTables
+import com.openlattice.postgres.PostgresTable.DATA
+import org.slf4j.LoggerFactory
+
+class AddOriginIdToDataPrimaryKey(private val toolbox: Toolbox) : Upgrade {
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(AddOriginIdToDataPrimaryKey::class.java)
+    }
+
+    override fun getSupportedVersion(): Long {
+        return Version.V2019_10_03.value
+    }
+
+    override fun upgrade(): Boolean {
+
+        val pkey = (PostgresDataTables.buildDataTableDefinition().primaryKey + ORIGIN_ID).joinToString(",") { it.name }
+
+        val dropPkey = "ALTER TABLE ${DATA.name} DROP CONSTRAINT ${DATA.name}_pkey"
+        val updatePkey = "ALTER TABLE ${DATA.name} ADD PRIMARY KEY ($pkey)"
```

Comment:
I would also rename the index corresponding to the primary key and delete it *after* the new index has been created.

Reply:
I don't think postgres will allow you to have multiple primary keys at once

Reply:
although if there's a way to rename the constraint and remove its primary key status then that would be ideal

Reply:
Primary key is a constraint, the index is separate from the constraint.